scripts/globals/magic.lua

```diff
@@ -468,141 +317,66 @@ params.skillType = $3
+    local magicHitRate = xi.combat.magicHitRate.calculateMagicHitRate(magicAcc, magicEva)
+    local resistRate   = xi.combat.magicHitRate.calculateResistRate(actor, target, xi.skill.NONE, element, magicHitRate, 0)
 
-    return getMagicResist(player, target, xi.skill.NONE, element, magicHitRate)
-end
-
-function getMagicHitRate(caster, target, skillType, element, percentBonus, bonusAcc)
-    -- resist everything if real magic shield is active (see effects/magic_shield)
-    if target:hasStatusEffect(xi.effect.MAGIC_SHIELD) then
-        local magicshieldsub = target:getStatusEffect(xi.effect.MAGIC_SHIELD)
-
-        if magicshieldsub:getSubPower() == 0 then
-            return 0
-        end
-    end
-
-    if bonusAcc == nil then
-        bonusAcc = 0
-    end
-
-    local magicacc = caster:getMod(xi.mod.MACC) + caster:getILvlMacc()
-
-    -- Get the base acc (just skill + skill mod (79 + skillID = ModID) + magic acc mod)
-    if skillType ~= 0 then
-        magicacc = magicacc + caster:getSkillLevel(skillType)
-    else
-        -- for mob skills / additional effects which don't have a skill
-        magicacc = magicacc + utils.getSkillLvl(1, caster:getMainLvl())
-    end
-
-    local resMod = 0 -- Some spells may possibly be non elemental, but have status effects.
-    if element ~= xi.element.NONE then
-        resMod = target:getMod(xi.magic.resistMod[element])
-
-        -- Add acc for elemental affinity accuracy and element specific accuracy
-        local affinityBonus = AffinityBonusAcc(caster, element)
-        local elementBonus = caster:getMod(spellAcc[element])
-
-        bonusAcc = bonusAcc + affinityBonus + elementBonus
-    end
-
-    magicacc = magicacc + caster:getMerit(xi.merit.MAGIC_ACCURACY)
-
-    magicacc = magicacc + caster:getMerit(xi.merit.NIN_MAGIC_ACCURACY)
-
-    -- Base magic evasion (base magic evasion plus resistances(players), plus elemental defense(mobs)
-    local magiceva = target:getMod(xi.mod.MEVA) + resMod
-
-    magicacc = magicacc + bonusAcc
-
-    -- Add macc% from food
-    local maccFood = magicacc * (caster:getMod(xi.mod.FOOD_MACCP) / 100)
-    magicacc = magicacc + utils.clamp(maccFood, 0, caster:getMod(xi.mod.FOOD_MACC_CAP))
-
-    return calculateMagicHitRate(magicacc, magiceva, percentBonus, caster:getMainLvl(), target:getMainLvl())
-end
-
--- Returns resistance value from given magic hit rate (p)
-function getMagicResist(caster, target, skill, element, magicHitRate)
-    return xi.combat.magicHitRate.calculateResistRate(caster, target, skill, element, magicHitRate, 0)
+    return resistRate
 end
 
 -- Returns the amount of resistance the target has to the given effect
 local effectToResistanceMod =
 {
     [xi.effect.SLEEP_I      ] = xi.mod.SLEEPRES,
     [xi.effect.SLEEP_II     ] = xi.mod.SLEEPRES,
-    [xi.effect.LULLABY      ] = xi.mod.LULLABYRES,
+    [xi.effect.LULLABY      ] = xi.mod.SLEEPRES,
```

Comment:
I know we found an NM that was resistant to lullaby and not sleep, do we want to make a TODO or a note here that it needs another look into at some point?

Reply:
by resist, i assume you mean immunity, of a hight dark resistance rank and not a job trait trigger.

Moving forward, we are going to use the immunity system for light and dark instead of te regular sleep immunity.

As for this, all of this here is going to be nuked in favor of the new magic system. This is just not being handled properly